### PR TITLE
OPENTOK-47982: Add updated e2ee api

### PR DESCRIPTION
**Pull request by maikthomas**
#### What is this PR doing?
- Update usage of e2ee API.
- Use custom version of opentok-angular to support connection options

#### How should this be manually tested?
- checkout branch and run npm install
- run locally
- create a new room, enabling e2ee with a secret longer than 8 characters
- join room with same secret from 2nd tab, ensure all works ok
- join with different secret, check it doesn't work
- join with e2ee false and check it doesn't work

#### What are the relevant tickets?
OPENTOK-47982

### CODE REVIEW

**Review by v-kpheng — APPROVED** [review]
LGTM! 💪 🚀

File: src/js/controllers.js [hunk 1]
```diff
@@ -456,55 +446,60 @@ angular.module('opentok-meet').controller('RoomCtrl', ['$scope', '$http', '$wind
...
-            $scope.reconnecting = isReconnecting;
-          });
-        };
-        if ((session.is && session.is('connected')) || session.connected) {
-          connectDisconnect(true);
-        }
-        $scope.session.on('sessionConnected', connectDisconnect.bind($scope.session, true));
-        $scope.session.on('sessionDisconnected', connectDisconnect.bind($scope.session, false));
-        $scope.session.on('archiveStarted archiveStopped', (event) => {
+      const sessionOptions = e2eeSecret ? { encryptionSecret: e2eeSecret } : {};
```
[the remark this answers is not in the document]
Comment (maikthomas) [reply src/js/controllers.js:449]: good idea

File: src/js/append-query-param-to-url.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const appendQueryParamToUrl = (queryString, url) => {
+  const delimiter = url.includes('?') ? '&' : '?';
+  return url + delimiter + queryString;
```
[the remark this answers is not in the document]
Comment (maikthomas) [reply src/js/append-query-param-to-url.js]: updated

**Review by v-kpheng — APPROVED** [review]
LGTM! 💪 🚀

**Pull request merged**